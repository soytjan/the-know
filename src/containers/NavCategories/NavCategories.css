@import url("https://fonts.googleapis.com/css?family=Open+Sans:300,400|Poiret+One");
.NavCategories {
  width: 800px;
  display: flex;
  justify-content: space-between; }

.a-category button {
  font-size: 1.2em;
  width: 160px;
  padding: 5px; }
  .a-category button:hover {
    background-color: #7848b5; }

.a-category.active button {
  color: #383838;
  background-color: #BEECE9;
  font-weight: 400; }
