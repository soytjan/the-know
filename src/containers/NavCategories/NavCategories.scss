@import "../../variables.scss";

.NavCategories {
  width: 800px;
  display: flex;
  justify-content: space-between;

}

.a-category {
  button {
    color: $nav-font-btn-color;
    font-size: 1.2em;
    width: 160px;
    padding: 5px;
    border: 2px solid $nav-border-color;
    background-color: $white;
  }
}

.a-category.active {
  button {
    color: $black;
    background-color: $nav-category-active-bg;
    font-weight: 400;
  }
}