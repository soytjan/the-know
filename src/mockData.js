export const mockEventData = {
  "last_item": null,
  "total_items": "9621",
  "first_item": null,
  "page_number": "1",
  "page_size": "10",
  "page_items": null,
  "search_time": "0.049",
  "page_count": "963",
  "events": {
    "event": [
      {
        "watching_count": null,
        "olson_path": "America/Denver",
        "calendar_count": null,
        "comment_count": null,
        "region_abbr": "CO",
        "postal_code": null,
        "going_count": null,
        "all_day": "0",
        "latitude": "39.7392358",
        "groups": null,
        "url": "http://denver.eventful.com/events/thursday-work-day-july-5-multiple-neighborhoods-/E0-001-111209195-8?utm_source=apis&utm_medium=apim&utm_campaign=apic",
        "id": "E0-001-111209195-8",
        "privacy": "1",
        "city_name": "Denver",
        "link_count": null,
        "longitude": "-104.990251",
        "country_name": "United States",
        "country_abbr": "USA",
        "region_name": "Colorado",
        "start_time": "2018-07-05 09:00:00",
        "tz_id": null,
        "description": " <strong>Sponsored by:</strong> Extreme Community Makeover<br> <strong>Minimum age:</strong> 14<br> Extreme Community Makeover (ECM) is designed to invite volunteers into a partnership with residents of specific Denver neighborhoods to complete home and neighborhood improvement projects. These projects include graffiti removal, landscaping, painting, yard work, or other projects identified by the . . .",
        "modified": "2018-02-19 06:12:10",
        "venue_display": "0",
        "tz_country": null,
        "performers": null,
        "title": "Thursday Work Day, July 5 - Multiple Neighborhoods",
        "venue_address": "Denver, Colorado, United States",
        "geocode_type": "EVDB Geocoder",
        "tz_olson_path": null,
        "recur_string": null,
        "calendars": null,
        "owner": "evdb",
        "going": null,
        "country_abbr2": "US",
        "image": null,
        "created": "2018-02-01 04:56:32",
        "venue_id": "V0-001-000179395-7",
        "tz_city": null,
        "stop_time": null,
        "venue_name": "Denver, Colorado, United States",
        "venue_url": "http://denver.eventful.com/venues/denver-colorado-united-states-/V0-001-000179395-7?utm_source=apis&utm_medium=apim&utm_campaign=apic"
      }
    ]
  }
}

export const mockCleanEventData = [
  {
    title: "Thursday Work Day, July 5 - Multiple Neighborhoods",
    description: " <strong>Sponsored by:</strong> Extreme Community Makeover<br> <strong>Minimum age:</strong> 14<br> Extreme Community Makeover (ECM) is designed to invite volunteers into a partnership with residents of specific Denver neighborhoods to complete home and neighborhood improvement projects. These projects include graffiti removal, landscaping, painting, yard work, or other projects identified by the . . .",
    category: "event",
    venueName: "Denver, Colorado, United States", 
    region: "CO",
    postalCode: null,
    city: "Denver",
    startTime: "2018-07-05 09:00:00",
    venueAddress: "Denver, Colorado, United States",
    image: null,
    url: "http://denver.eventful.com/events/thursday-work-day-july-5-multiple-neighborhoods-/E0-001-111209195-8?utm_source=apis&utm_medium=apim&utm_campaign=apic",
    id: "E0-001-111209195-8"
  }
]