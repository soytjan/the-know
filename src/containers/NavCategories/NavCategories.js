import React, { Component } from 'react';
import { connect } from 'react-redux';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';
import { fetchAndCleanCategoryEventData } from '../../helper';
import { addEvents } from '../../actions/';
import './NavCategories.css';

export class NavCategories extends Component {
  componentDidMount = async () => {
    const { location } = this.props;
    
    await this.getAndStoreEventsData(location);
  }

  getAndStoreEventsData = async (location) => {
    const { addEvents } = this.props;

    const musicEvents = await fetchAndCleanCategoryEventData('music', location);
    addEvents(musicEvents, 'music');
    const foodEvents = await fetchAndCleanCategoryEventData('food', location);
    addEvents(foodEvents, 'food');
    const cultureEvents = await fetchAndCleanCategoryEventData('culture', location);
    addEvents(cultureEvents, 'culture');
    const nightlifeEvents = await fetchAndCleanCategoryEventData('nightlife', location);
    addEvents(nightlifeEvents, 'nightlife');
  }

  render() {
    return (
      <nav className='NavCategories'>
        <NavLink to='/home/music' activeClassName="active" className='a-category'>
          <button>
            MUSIC
          </button>
        </NavLink>
        <NavLink to='/home/food' activeClassName="active" className='a-category'>
          <button>
            FOOD
          </button>
        </NavLink>
        <NavLink to='/home/culture' activeClassName="active" className='a-category'>
          <button>
            CULTURE
          </button>
        </NavLink>
        <NavLink to='/home/nightlife' activeClassName="active" className='a-category'>
          <button>
            NIGHTLIFE
          </button>
        </NavLink>  
      </nav>
    );
  }
}

NavCategories.propTypes = {
  location: PropTypes.object,
  addEvents: PropTypes.func
};

export const mapStateToProps = (state) => ({
  location: state.location
});

export const mapDispatchToProps = (dispatch) => ({
  addEvents: (events, category) => dispatch(addEvents(events, category))
});

export default connect(mapStateToProps, mapDispatchToProps)(NavCategories);