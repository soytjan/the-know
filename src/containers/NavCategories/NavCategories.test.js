import React from 'react';
import { shallow } from 'enzyme';
import { mockEventData, mockCleanGeocodeData } from '../../mockData';
import { NavCategories, mapStateToProps, mapDispatchToProps } from './NavCategories';

describe('NavCategories', () => {
  let renderedComponent;
  let mockAddEvents;

  beforeEach(() => {
    window.fetch = jest.fn().mockImplementation(() => Promise.resolve({
      status: 200,
      json: () => Promise.resolve(mockEventData)
    }));
    mockAddEvents = jest.fn();
    renderedComponent = shallow(
      <NavCategories 
        addEvents={mockAddEvents}
        location={mockCleanGeocodeData}
      />
    );
  });

  it('should match snapshot', () => {
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should map to the store correctly', () => {
    const mockStore = {
      location: mockCleanGeocodeData
    };
    const mapped = mapStateToProps(mockStore);

    expect(mapped).toEqual(mockStore);
    expect(mapped.location).toEqual(mockCleanGeocodeData);
  });

  it('should call the dispatch function when addEvents is called from MDTP', () => {
    const mockDispatch = jest.fn();
    const mapped = mapDispatchToProps(mockDispatch);
    mapped.addEvents();

    expect(mockDispatch).toHaveBeenCalled();
  });
});