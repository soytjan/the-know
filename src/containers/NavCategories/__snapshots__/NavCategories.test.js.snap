// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavCategories should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavCategories
    addEvents={[Function]}
    location={
        Object {
            "address": "Denver, CO, USA",
            "coordinates": Object {
              "lat": 39.7392358,
              "lng": -104.990251,
            },
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          className="a-category"
          to="/home/music"
>
          <button>
                    MUSIC
          </button>
</NavLink>,
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          className="a-category"
          to="/home/food"
>
          <button>
                    FOOD
          </button>
</NavLink>,
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          className="a-category"
          to="/home/culture"
>
          <button>
                    CULTURE
          </button>
</NavLink>,
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          className="a-category"
          to="/home/nightlife"
>
          <button>
                    NIGHTLIFE
          </button>
</NavLink>,
      ],
      "className": "NavCategories",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": <button>
            MUSIC
</button>,
          "className": "a-category",
          "to": "/home/music",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "MUSIC",
          },
          "ref": null,
          "rendered": "MUSIC",
          "type": "button",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": <button>
            FOOD
</button>,
          "className": "a-category",
          "to": "/home/food",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "FOOD",
          },
          "ref": null,
          "rendered": "FOOD",
          "type": "button",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": <button>
            CULTURE
</button>,
          "className": "a-category",
          "to": "/home/culture",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "CULTURE",
          },
          "ref": null,
          "rendered": "CULTURE",
          "type": "button",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": <button>
            NIGHTLIFE
</button>,
          "className": "a-category",
          "to": "/home/nightlife",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "NIGHTLIFE",
          },
          "ref": null,
          "rendered": "NIGHTLIFE",
          "type": "button",
        },
        "type": [Function],
      },
    ],
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="a-category"
            to="/home/music"
>
            <button>
                        MUSIC
            </button>
</NavLink>,
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="a-category"
            to="/home/food"
>
            <button>
                        FOOD
            </button>
</NavLink>,
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="a-category"
            to="/home/culture"
>
            <button>
                        CULTURE
            </button>
</NavLink>,
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="a-category"
            to="/home/nightlife"
>
            <button>
                        NIGHTLIFE
            </button>
</NavLink>,
        ],
        "className": "NavCategories",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": <button>
              MUSIC
</button>,
            "className": "a-category",
            "to": "/home/music",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "MUSIC",
            },
            "ref": null,
            "rendered": "MUSIC",
            "type": "button",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": <button>
              FOOD
</button>,
            "className": "a-category",
            "to": "/home/food",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "FOOD",
            },
            "ref": null,
            "rendered": "FOOD",
            "type": "button",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": <button>
              CULTURE
</button>,
            "className": "a-category",
            "to": "/home/culture",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "CULTURE",
            },
            "ref": null,
            "rendered": "CULTURE",
            "type": "button",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": <button>
              NIGHTLIFE
</button>,
            "className": "a-category",
            "to": "/home/nightlife",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "NIGHTLIFE",
            },
            "ref": null,
            "rendered": "NIGHTLIFE",
            "type": "button",
          },
          "type": [Function],
        },
      ],
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
